### ISSUE TITLE
[UI] Convert all Class-based components to functional components 

### ISSUE THREAD

**Issue by Rajdip019** [meshery/meshery#8668]
### Current Behavior
A number of components are class-based. 

### Desired Behavior
Convert all the Class-based components to new updated functional components and change the required functions accordingly.

### Implementation
Convert all the Class-based components to new updated functional components. It is important here to convert a class-based component to a functional component you may need to convert some functions as well. Check if the function is used in other components or not.

Also, make sure to use the `hooks` while converting them. For example, you need to covert `withNotify` with `useNotification` hook. Please check if a hook for functional components is available and use that.

### Acceptance Tests
Make sure the component you have converted should work the same as before and try to upload a video of the working component.

### List of components we need to convert 👇🏽 
- [ ] Header - `/ui/components/Header.js`
- [ ] LoadTestTimerDialog - `/ui/components/load-test-timer-dialog.js`
- [ ] MeshAdapterConfigComponent - `/ui/components/MeshAdapterConfigComponent.js`
- [ ] MesheryAdapterPlayComponent - `/ui/components/MesheryAdapterPlayComponent.js`
- [ ] MesheryChart - `/ui/components/MesheryChart.js`
- [ ] MesheryChartDialog - `/ui/components/MesheryChartDialog.js`
- [ ] MesheryNotification - `/ui/components/MesheryNotification.js`
- [ ] MesheryPlayComponent - `/ui/components/MesheryPlayComponent.js`
- [ ] MesheryResultDialog - `/ui/components/MesheryResultDialog.js`
- [ ] MesherySettings - `/ui/components/MesherySettings.js`
- [ ] MesherySettingsPerformanceComponent - `/ui/components/MesherySettingsPerformanceComponent.js`
- [ ] Navigator - `/ui/components/Navigator.js`
- [ ] PromptComponent - `/ui/components/PromptComponent.js`
- [ ] ReactSelectWrapper - `/ui/components/ReactSelectWrapper.js`
- [ ] User - `/ui/components/User.js`
- [ ] UserPreference - `/ui/components/UserPreferences.js`
- [ ] MesheryPerformanceComponent - `/ui/components/MesheryPerformance/index.js`
- [ ] Index - `/ui/pages/index.js`
- [ ] Settings - `/ui/pages/settings.js`
- [ ] SMIResults - `/ui/pages/smi_results.js`
- [ ] NewFilters - `/ui/pages/configuration/filters.js`
- [ ] RemoteExtension - `/ui/pages/extension/[...component].js`
- [ ] Connections - `/ui/pages/management/connections.js`
- [ ] Manage - `/ui/pages/management/index.js`
- [ ] UserPref - `/ui/pages/user/preferences.js`

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

**Comment by ShatilKhan** [comment]
I'd like to work on this issue

**Comment by EraKin575** [comment]
@Rajdip019 you can co assign me for working on a component

**Comment by sudhanshutech** [comment]
@Rajdip019 keep also track of this issue https://github.com/meshery/meshery/issues/7187
Is quite same

**Comment by Rajdip019** [comment]
> @Rajdip019 keep also track of this issue #7187 Is quite same

Got it. Will keep track of that.

**Comment by Rajdip019** [comment]
> @Rajdip019 you can co assign me for working on a component

Yes sure will start co-assigning with people after today's development meet.

**Comment by Cvr421** [comment]
Hii Rajdeep i would like to work on this.

**Comment by Layak-Ali** [comment]
Hey @Rajdip019 , i would like to contribute to this issue.

**Comment by sadamhussain-m** [comment]
Will U assign any one component for me to convert?

**Comment by Rajdip019** [comment]
Sure see the components and let me know which components you guys want to convert. I will suggest starting form the small ones and moving up to the bigger ones. @sadamhussain-m @Layak-Ali @Cvr421

**Comment by csengineer1990** [comment]
hi, I am a new newcomer. I want to contribute to this issue. Please guide me so I can complete the assigned issue.?

**Comment by Rajdip019** [comment]
> @Rajdip019 I'm open to accept any small component to work on from your suggestion.

See the components and feel free to choose one whichever you feel confident in.

**Comment by anniepauline** [comment]
Willing to work on it!

**Comment by anniepauline** [comment]
> > @Rajdip019 I'm open to accept any small component to work on from your suggestion.
> 
> See the components and feel free to choose one whichever you feel confident in.

Can we start working on it, without you having to assign the issue?

**Comment by ShatilKhan** [comment]
If there's no problems with assigning I guess we can start working on it.
If anyone starts working, please let us know via comment which components you are working on.

**Comment by Cvr421** [comment]
Hii @Rajdip019 i would like to work on Mesherysettting component /ui/components/MesherySettings.js

**Comment by EraKin575** [comment]
i would like to work on RemoteExtensions component @Rajdip019

**Comment by Rajdip019** [comment]
> I've selected these three pages Index - `/ui/pages/index.js` Settings -` /ui/pages/settings.js` SMIResults - `/ui/pages/smi_results.js`
> 
> And this component: PromptComponent - `/ui/components/PromptComponent.js`
> 
> I will be working on these, I'll start with index.js for now. Although if anyone else wants to work on any of these, do let me know.

Let's go two at a time. I have made and issue for you!

**Comment by Rajdip019** [comment]
> i would like to work on RemoteExtensions component @Rajdip019

Hey, sure but can you focus first on finishing the Dashboard component? Then you can surely start with these.

**Comment by Rajdip019** [comment]
Hey, @ishashankmittal do you have the local setup done? and can you confirm you properly understand the issue?  So that I can go ahead and assign some components to you!

**Comment by sadamhussain-m** [comment]
Could you assign me the below one:

/ui/pages/management/index.js

**Comment by ishashankmittal** [comment]
> Hey, @ishashankmittal do you have the local setup done? and can you confirm you properly understand the issue? So that I can go ahead and assign some components to you!

Yes, I have the local Setup done and it is up and running. I also understand the issue properly.
Could you assign me /ui/components/MesheryNotification.js ?

**Comment by SoNiC-HeRE** [comment]
Hi @Rajdip019 could you assign me ```/ui/components/Header.js```?

**Comment by Rajdip019** [comment]
Do you have local setup running? @SoNiC-HeRE

**Comment by mukeshkuiry** [comment]
@Rajdip019 I would like to work on 
`/ui/pages/user/preferences.js`

**Comment by Rajdip019** [comment]
Does this component has any dependencies like functions that are for class based components or any other class based component attached to it?

@mukeshkuiry

**Comment by mukeshkuiry** [comment]
Yes there is component `/ui/components/UserPreferences.js` is linked

so assign both to me, will change in both.

**Comment by 20RitikSingh** [comment]
I want work on this issue.

**Comment by Rajdip019** [comment]
Sure @20RitikSingh make sure you have local system up and running and the choose two component that you feel you can work on.

**Comment by suzy-g38** [comment]
@Rajdip019 I want to work on this issue. Can you suggest to me a page that you guys are not working on or not assigned? You can give me any page that you guys are not working on.

**Comment by Rajdip019** [comment]
> @Rajdip019 I want to work on this issue. Can you suggest to me a page that you guys are not working on or not assigned? You can give me any page that you guys are not working on.

Make sure to have local server running and also I would suggest picking a component to page you feel confident. Choose one which doesn't have a [ in progress ] mark beside it.

**Comment by suzy-g38** [comment]
> > @Rajdip019 I want to work on this issue. Can you suggest to me a page that you guys are not working on or not assigned? You can give me any page that you guys are not working on.
> 
> Make sure to have local server running and also I would suggest picking a component to page you feel confident. Choose one which doesn't have a [ in progress ] mark beside it.

@Rajdip019 I would like to choose MesheryProgressBar.js and MeshAdapterConfigComponent.js under /ui/components for now.

**Comment by mukeshkuiry** [comment]
@Rajdip019 `/ui/components/User.js` assign as it's not started by anyone.

**Comment by nebula-aac** [comment]
@Rajdip019 I have partially completion on `Header.js`, I can complete this. It would be better to update the current issue with whom are working on it so everyone is aware.

**Comment by EraKin575** [comment]
@Rajdip019 can you assign me a component to change?

**Comment by Rajdip019** [comment]
> @Rajdip019 I have partially completion on `Header.js`, I can complete this. It would be better to update the current issue with whom are working on it so everyone is aware.

Thanks for the clarification. You can tick off `Header.js` for this list as well when it's merged.

**Comment by Rajdip019** [comment]
> @Rajdip019 will you assign me a component to change?

Choose one you are compatible with and not in progress. And I will make a new issue and assign you!

**Comment by EraKin575** [comment]
@Rajdip019  please assign me the RemoteExtensions component

**Comment by EraKin575** [comment]
@Rajdip019 on the problem description, I think you should also mention using the useRouter hook for page transitions

**Comment by Rajdip019** [comment]
> @Rajdip019 on the problem description, I think you should also mention using the useRouter hook for page transitions

Thanks for the suggestion I have added that.

**Comment by EraKin575** [comment]
@Rajdip019 please assign me LoadTestTimeConfig and MeshAdapterConfigComponent

**Comment by csengineer1990** [comment]
Hi, @Rajdip019, can I work on 'PromptComponent' - '/ui/components/PromptComponent.js'? It looks like it's not too difficult for a newcomer, and I'm wondering if it's available.?

**Comment by mukeshkuiry** [comment]
Is there any component left? @Rajdip019

**Comment by Cvr421** [comment]
@Rajdip019  i would like work on UserPreference - /ui/components/UserPreferences.js component . Is it assigned to someone  or not . if not can i work on this .

**Comment by abhijeetgauravm** [comment]
hey @Rajdip019 NewFilters - /ui/pages/configuration/filters.js I would like to work on this.

**Comment by EraKin575** [comment]
@Rajdip019 please assign me LoadTestTimeConfig and MeshAdapterConfigComponent

**Comment by Rajdip019** [comment]
> Is there any component left? @Rajdip019

Choose one on which no one is working and raise an issue.

**Comment by laviedegeorge** [comment]
@Rajdip019  still need help? I am available.

**Comment by dhiraj0911** [comment]
@Rajdip019 would like to work
If any file is remaining can you please let me know

**Comment by Ishwar-Zatke** [comment]
@Rajdip019 I would like to work on PromptPerformance, it is remaing to change, please let me know

**Comment by mukeshkuiry** [comment]
@Rajdip019 
MesheryChart assign me this

**Comment by sudhanshutech** [comment]
@Rajdip019 please mark as completed the checkboxes as they are merged

**Comment by Rajdip019** [comment]
> @Rajdip019 please mark as completed the checkboxes as they are merged

Yes sure. doing that.

**Comment by Cvr421** [comment]
@Rajdip019 is their any component remaining to work on . Please let me know .

**Comment by Rajdip019** [comment]
Choose something that doesn't have a [ In progress ] tag beside it.

**Comment by SoNiC-HeRE** [comment]
> Do you have local setup running? @SoNiC-HeRE

Sorry for the late reply, Yes i have got the local setup working; it would be great if you could assign me the component or any other component that's left to do

**Comment by Cvr421** [comment]
@Rajdip019 i would like to work on this  Navigator - /ui/components/Navigator.js

**Comment by Cvr421** [comment]
> Choose something that doesn't have a [ In progress ] tag beside it.

@Rajdip019  i would like to work on this  Navigator - /ui/components/Navigator.js

**Comment by csengineer1990** [comment]
Hi, @Rajdip019  is there any component left to work on?

**Comment by iArchitSharma** [comment]
@Rajdip019 Can you please assign me `/ui/components/MesherySettingsPerformanceComponent.js` if its not already taken

**Comment by aabidsofi19** [comment]
@Rajdip019 i think MesheryNotification.js is only one that got refactored in latest updates

**Comment by coder12git** [comment]
@Rajdip019 I want to work, for rest other components, that are left. Let me know which components are left now :)

**Comment by Pushkarm029** [comment]
@Rajdip019 @saurabh100ni Is this issue still in need of work? If yes, I want to work on these components: MesheryChartDialog: /ui/components/MesheryChartDialog.js

**Comment by coder12git** [comment]
// @iArchitSharma

**Comment by iArchitSharma** [comment]
@coder12git  I think all the components have already been assigned as of now

**Comment by iArchitSharma** [comment]
It's been a quiet while, and it's not completed. Let's create an issue for these components and get them assigned. I have rechecked; some of them were already converted, so I have checkmarked those. Also, I've added a new component to the list

**Comment by giteshsarvaiya** [comment]
#10344
may I know in which path can I see the "header.js" component?

**Comment by iArchitSharma** [comment]
@girishranganathan https://github.com/meshery/meshery/blob/master/ui/components/Header.js

so you are working on it?

**Comment by giteshsarvaiya** [comment]
Yess, I am working on it, thanks for the link,  I meant the url path where the component with promptComponent as a child was present so that I can test the changes... (I found it, thanks again)
